File: filmogophery-front/src/interface/movie.ts
import { Genre } from "./genre";
import { Review } from "./review";

export type Movie = {
  id: number;
  title: string;
  overview: string;
  releaseDate: string;
  runtimeMinutes: number;
  posterURL: string | null;
  tmdbID: number;
  genres: Genre[];
};

// Movie を継承
export type MovieDetail = Movie & {
  voteAverage: number; // Min:0.0, Max:5.0
  voteCount: number;
  series: Series | null;
  review: Review | null;
};

type Series = {
  name: string;
  posterURL: string | null;
};

export type SearchMovie = {
  tmdbID: number;
  title: string;
  overview: string;
  popularity: number;
  posterURL: string;
  releaseDate: string;
  voteAverage: number;
  voteCount: number;
  genres: string[];
};

File: api-echo/internal/app/types/movie.go
package types

import (
	"filmogophery/internal/pkg/constant"
)

type (
	Movie struct {
		ID             int32         `json:"id"`
		Title          string        `json:"title"`
		Overview       string        `json:"overview"`
		ReleaseDate    constant.Date `json:"releaseDate"`
		RuntimeMinutes int32         `json:"runtimeMinutes"`
		PosterURL      *string       `json:"posterURL"`
		TmdbID         int32         `json:"tmdbID"`
		Genres         []Genre       `json:"genres"`
	}
	MovieDetail struct {
		VoteAverage float32 `json:"voteAverage"`
		VoteCount   int32   `json:"voteCount"`
		Series      *Series `json:"series"`
		Review      *Review `json:"review"`
		Movie
	}

	SearchMovie struct {
		TmdbID      int32    `json:"tmdbID"`
		Title       string   `json:"title"`
		Overview    string   `json:"overview"`
		Popularity  int32    `json:"popularity"`
		PosterURL   string   `json:"posterURL"`
		ReleaseDate string   `json:"releaseDate"`
		VoteAverage int32    `json:"voteAverage"`
		VoteCount   int32    `json:"voteCount"`
		Genres      []string `json:"genres"`
	}
)
